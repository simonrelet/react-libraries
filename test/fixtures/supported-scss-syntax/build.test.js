'use strict'

const fs = require('fs-extra')
const path = require('path')
const { cleanFixture } = require('../../fixtureContent')
const { callScriptInPackage } = require('../../scriptExecution')
const packageJSON = require('./package.json')

describe(`${packageJSON.name} build`, () => {
  beforeAll(async () => {
    await cleanFixture(__dirname)
  })

  it('should not fail', async () => {
    const result = await callScriptInPackage(__dirname, 'build')
    expect(result).toMatchSnapshot()
  })

  it('should compile to CSS', async () => {
    const compiledCSS = await fs.readFile(
      path.resolve(__dirname, packageJSON.style),
      'utf8',
    )

    expect(compiledCSS).toMatchSnapshot()
  })
})
