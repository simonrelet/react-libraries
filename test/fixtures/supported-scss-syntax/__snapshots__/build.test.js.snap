// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supported-scss-syntax build should compile to CSS 1`] = `".simple-class{background:red}.simple-class:before{background:#000}.nested-selectors,.simple-class:hover{background:green}.nested-selectors:focus{background:#000}.nested-selectors--modifier{background:#ff0}.nested-selectors:after{background:purple}.with-mixin-1{background:#00f;color:#000}.with-mixin-2{background:#00f;color:#fff}.with-placeholder-1,.with-placeholder-2{background:green}.with-placeholder-1{color:#000}.with-placeholder-2{color:#fff}"`;

exports[`supported-scss-syntax build should not fail 1`] = `
Exit code: 0

--- Output ---
CommonJS build successful.
SASS build successful.
--------------
`;
